### PR TITLE
Slack notifications

### PR DESCRIPTION
**Pull request by atalive**
Current channel #test-bot https://liveintent.slack.com/archives/C01GMD8KBV0
1. added a module to post messages on slack
2. modified circleci config to post to slack after running the tests

### CODE REVIEW

**Review by melgenek — APPROVED** [review]

**Comment by jankoulaga** [comment]
@atalive @melgenek why is this approved if it cannot even be merged? The diff looks horrible because of merge problems.

**Comment by atalive** [comment]
I am not sure why it is approved. But I need to clean up first.

**Comment by jankoulaga** [comment]
@atalive when will this be ready for review?

**Comment by jankoulaga** [comment]
@atalive ready for review?

File: fixtures/scenarios/ad-hoc/live-pixel-check.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "tags": ["toRun", "live-pixel"],
```
Comment (jankoulaga) [inline fixtures/scenarios/ad-hoc/live-pixel-check.json:2]: let's remove these `toRun` tags... they're not really useful

File: package.json [hunk 2]
```diff
@@ -4,7 +4,8 @@
   "description": "e2e test project",
   "scripts": {
     "test": "mocha test/scenario-runner.spec.js",
-    "bootstrap-scenario": "node src/scenarios/scenario-bootstrap.js"
+    "bootstrap-scenario": "node src/scenarios/scenario-bootstrap.js",
+    "posttest": "node reports/slack-bot.js"
```
Comment (jankoulaga) [inline package.json:8]: ehm... i'd call this `slack-report` and i'd rename the file, as it's not a slack bot.

File: reports/message-generator.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        })
+    })
+
+    return _.uniqWith(failedTests, _.isEqual)
+}
+
+function blocks() {
+    let blocks = []
+
+    const parts = require('./message-parts')
```
Comment (jankoulaga) [inline reports/message-generator.js:37]: why can't you require this on top? I don't think it needs to be in the body of a function.

File: reports/message-generator.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    const failed = failures()
+
+    if (_.isEmpty(failed)) {
+        blocks.push(parts["title-passed"])
+    } else {
+        blocks.push(parts["title-failed"])
+        Array.prototype.push.apply(blocks, failed)
```
Comment (jankoulaga) [inline reports/message-generator.js:44]: isn't this just `block.concat(failed)`?

File: reports/slack-bot.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+            text: 'Fallback text',
+            blocks: message.blocks()
+        })
+        console.log(result)
+    } catch (error) {
+        console.error('Failed to post to slack channel', error)
+    }
+}
+
+module.exports = postMessage()
```
Comment (jankoulaga) [inline reports/slack-bot.js:22]:
```suggestion
module.exports = postMessage()

```

File: reports/slack-bot.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+        logLevel: 'debug'
+    })
+    try {
+        const result = await webClient.chat.postMessage({
+            token: process.env.SLACK_API_TOKEN,
+            channel: process.env.SLACK_DEFAULT_CHANNEL,
+            text: 'Fallback text',
+            blocks: message.blocks()
+        })
+        console.log(result)
```
Comment (jankoulaga) [inline reports/slack-bot.js:16]: Please use the logger here, and in the error part.

File: reports/message-parts.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
+module.exports = {
+    "title-passed": {
```
Comment (jankoulaga) [inline reports/message-parts.js:2]: If this is a js file, then we should follow the convention of using single quotes.

File: reports/message-generator.js (hunk 4 above)
Comment (atalive) [reply reports/message-generator.js:44]: yes it is the same. i was just reading that push is way faster

File: package.json (hunk 2 above)
Comment (atalive) [reply package.json:8]: I put `posttest` then it will be executed automatically after npm test

**Review by jankoulaga — APPROVED** [review]

**Comment by jankoulaga** [comment]
@atalive shall we merge?

**Comment by atalive** [comment]
> @atalive shall we merge?

yes :)

**Comment by jankoulaga** [comment]
Go for it then! :)

**Comment by atalive** [comment]
> > @atalive shall we merge?
> 
> yes :)

i will just change the channel

**Pull request merged**